src/Control/Comonad/Trans/Env.hs

```diff
@@ -145,6 +149,22 @@ runEnv (EnvT e (Identity a)) = (e, a)
 runEnvT :: EnvT e w a -> (e, w a)
 runEnvT (EnvT e wa) = (e, wa)
 
+#if MIN_VERSION_base(4,9,0)
+instance (Eq e, Eq1 w) => Eq1 (EnvT e w) where
+  liftEq eq (EnvT e1 wa1) (EnvT e2 wa2) = e1 == e2 && liftEq eq wa1 wa2
+
+instance (Ord e, Ord1 w) => Ord1 (EnvT e w) where
+  liftCompare comp (EnvT e1 wa1) (EnvT e2 wa2) = case compare e1 e2 of
```

Comment:
I would prefer to implement this using `Ordering`'s `Semigroup` instance:

```hs
liftCompare comp (EnvT e1 wa1) (EnvT e2 wa2) = compare e1 e2 <> liftCompare comp wa1 wa2
```

This makes the symmetry with the `Eq1` instance much clearer.